livecheck/utils/portage.py

```diff
@@ -139,10 +139,10 @@ def catpkg_catpkgsplit(atom: str) -> tuple[str, str, str, str]:
 
 def get_first_src_uri(match: str, search_dir: str | None = None) -> str:
     try:
-        for uri in P.aux_get(match, ['SRC_URI'], mytree=search_dir):
-            for line in uri.split():
-                if line.startswith(('http://', 'https://')):
-                    return line
+        for line in P.aux_get(match, ['SRC_URI'], mytree=search_dir):
+            for uri in line.split():
+                if uri.startswith(('http://', 'https://', 'mirror://', 'ftp://')):
+                    return uri
```

Comment:
Use `chain` (`from itertools import chain`) and `next`:

```suggestion
        if (found_uri := next(
            (uri for uri in chain(*(x.split()
                                    for x in P.aux_get(match, ['SRC_URI'], mytree=search_dir)))
             if uri.startswith(('http://', 'https://', 'mirror://', 'ftp://'))), None)):
            return found_uri
```